# refentry-page-sequence.patch by Peter Eisentraut <petere@debian.org>
#
# Changes refentry element to produce a simple-page-sequence as root
# object when the refentry element is at the root of the document
# (bug #298163)

diff -urNad docbook-dsssl-1.79/print/dbrfntry.dsl /tmp/dpep.4ilCdq/docbook-dsssl-1.79/print/dbrfntry.dsl
--- docbook-dsssl-1.79/print/dbrfntry.dsl	2003-01-15 09:24:23.000000000 +0100
+++ /tmp/dpep.4ilCdq/docbook-dsssl-1.79/print/dbrfntry.dsl	2005-04-15 22:30:02.599246707 +0200
@@ -80,28 +80,46 @@
 ;;      (empty-sosofo)))
 
 (element refentry 
-  (make display-group
-    keep: %refentry-keep%
-    (if (or %refentry-new-page%
-	    (node-list=? (current-node) (sgml-root-element)))
-	(make simple-page-sequence
-	  page-n-columns: %page-n-columns%
-	  page-number-format: ($page-number-format$)
-	  use: default-text-style
-	  left-header:   ($left-header$)
-	  center-header: ($center-header$)
-	  right-header:  ($right-header$)
-	  left-footer:   ($left-footer$)
-	  center-footer: ($center-footer$)
-	  right-footer:  ($right-footer$)
-	  input-whitespace-treatment: 'collapse
-	  quadding: %default-quadding%
-	  ($refentry-title$)
-	  (process-children))
-	(make sequence
-	  ($refentry-title$)
-	  ($block-container$)))
-    (make-endnotes)))
+  (if (node-list=? (current-node) (sgml-root-element))
+      (make simple-page-sequence
+	page-n-columns: %page-n-columns%
+	page-number-format: ($page-number-format$)
+	use: default-text-style
+	left-header:   ($left-header$)
+	center-header: ($center-header$)
+	right-header:  ($right-header$)
+	left-footer:   ($left-footer$)
+	center-footer: ($center-footer$)
+	right-footer:  ($right-footer$)
+	input-whitespace-treatment: 'collapse
+	quadding: %default-quadding%
+	($refentry-title$)
+	(process-children)
+	(make-endnotes))
+      (if %refentry-new-page%
+	  (make display-group
+	    keep: %refentry-keep%
+	    (make simple-page-sequence
+	      page-n-columns: %page-n-columns%
+	      page-number-format: ($page-number-format$)
+	      use: default-text-style
+	      left-header:   ($left-header$)
+	      center-header: ($center-header$)
+	      right-header:  ($right-header$)
+	      left-footer:   ($left-footer$)
+	      center-footer: ($center-footer$)
+	      right-footer:  ($right-footer$)
+	      input-whitespace-treatment: 'collapse
+	      quadding: %default-quadding%
+	      ($refentry-title$)
+	      (process-children))
+	    (make-endnotes))
+	  (make display-group
+	    keep: %refentry-keep%
+	    (make sequence
+	      ($refentry-title$)
+	      ($block-container$))
+	    (make-endnotes)))))
 
 (define ($refentry-title$)
   (let* ((refmeta       (select-elements (children (current-node))
